### PR TITLE
feat(spf): add spf diag

### PR DESCRIPTION
**Pull request by maximeVaillant**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Add SPF diagnostic and pop up to fix it.
Fix SRV and MX pop up base on PU expected values

## Related

CMT-2340

### CODE REVIEW

**Review by anooparveti — CHANGES_REQUESTED** [review]
@maximeVaillant, as mentioned in https://github.com/ovh/manager/blob/master/CONTRIBUTING.md#translations.
Authors are not supposed to modify any content in translation files except `fr_FR`.
Can you please remove the modifications in other language files?

File: packages/manager/modules/exchange/src/domain/mx-autoconfig/domain-mx-autoconfig.controller.js [hunk 1]
```diff
@@ -25,26 +25,47 @@ export default class ExchangeDomainMxAutoconfigCtrl {
 
     this.$routerParams = wucExchange.getParams();
     this.domain = navigation.currentActionData;
+    this.domainDiag = {};
```
Comment (anooparveti) [inline packages/manager/modules/exchange/src/domain/mx-autoconfig/domain-mx-autoconfig.controller.js:28]:
```suggestion
    this.domainDialog = {};
```

File: packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  ) {
+    this.services = {
+      $scope,
+      wucExchange,
+      ExchangeDomains,
+      navigation,
+      messaging,
+      $translate,
+      exchangeStates,
+    };
```
Comment (anooparveti) [inline packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js:20]: **nitpick:** Unnecessary wrapper object.
```suggestion
      this.$scope = $scope;
      this.wucExchange = wucExchange;
      .
      .
      .
      this.exchangeStates = exchangeStates;
```

File: packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    this.$routerParams = wucExchange.getParams();
+    this.domain = navigation.currentActionData;
+    this.domainDiag = {};
```
Comment (anooparveti) [inline packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js:24]:
```suggestion
    this.domainDialog = {};
```

File: packages/manager/modules/exchange/src/domain/mx-autoconfig/domain-mx-autoconfig.controller.js (hunk 1 above)
Comment (maximeVaillant) [reply packages/manager/modules/exchange/src/domain/mx-autoconfig/domain-mx-autoconfig.controller.js:28]: domainDiag was not my choice, was already defined like that before.
For info, it's not mean Dialog but Diagnose ;)

File: packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js (hunk 3 above)
Comment (maximeVaillant) [reply packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js:24]: domainDiag was not my choice, was already defined like that before.
For info, it's not mean Dialog but Diagnose ;)

File: packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js (hunk 2 above)
Comment (maximeVaillant) [reply packages/manager/modules/exchange/src/domain/spf-autoconfig/domain-spf-autoconfig.controller.js:20]: It's same as packages/manager/modules/exchange/src/domain/srv-autoconfig/domain-srv-autoconfig.controller.js and packages/manager/modules/exchange/src/domain/mx-autoconfig/domain-mx-autoconfig.controller.js already on prod.
To keep simple code and homogeneity, I think it's better to keep it like that until a PR to improve globally code will be made.

File: packages/manager/modules/exchange/src/domain/srv-autoconfig/domain-srv-autoconfig.controller.js [hunk 4]
```diff
@@ -21,14 +21,43 @@ export default class ExchangeDomainSrvAutoconfigCtrl {
 
     this.$routerParams = wucExchange.getParams();
     this.domain = navigation.currentActionData;
+    this.domainDiag = {};
```
Comment (anooparveti) [inline packages/manager/modules/exchange/src/domain/srv-autoconfig/domain-srv-autoconfig.controller.js:24]:
```suggestion
    this.domainDialog = {};
```

Comment (maximeVaillant) [reply packages/manager/modules/exchange/src/domain/srv-autoconfig/domain-srv-autoconfig.controller.js:24]: domainDiag was not my choice, was already defined like that before.
For info, it's not mean Dialog but Diagnose ;)

**Review by anooparveti — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by sachinrameshn — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**